File: android/src/main/java/com/flutter/stripe/flutterstripe/FlutterStripePlugin.java
package com.flutter.stripe.flutterstripe;

import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

import com.stripe.android.Stripe;
import com.stripe.android.ApiResultCallback;
import com.stripe.android.model.Card;
import com.stripe.android.model.Token;

import java.util.HashMap;

/** FlutterStripePlugin */
public class FlutterStripePlugin implements FlutterPlugin, MethodCallHandler {
  static Registrar r;
  FlutterPluginBinding binding;

  /** Plugin registration. */
  public static void registerWith(Registrar registrar) {
    r = registrar;
    final MethodChannel channel = new MethodChannel(registrar.messenger(), "flutter_stripe");
	FlutterStripePlugin plugin = new FlutterStripePlugin();
    channel.setMethodCallHandler(plugin);
  }

  @Override
  public void onAttachedToEngine(FlutterPluginBinding b) {
    binding = b;
  }

  @Override
  public void onDetachedFromEngine(FlutterPluginBinding binding) {

  }

  @Override
  public void onMethodCall(MethodCall call, final Result result) {
    switch (call.method) {
      case "getPlatformVersion":
        result.success("Android " + android.os.Build.VERSION.RELEASE);
        break;
      case "getToken":
        HashMap<String, Object> arguments = (HashMap<String, Object>) call.arguments;
        Card.Builder builder = new Card.Builder(
                (String) arguments.get("cardNumber"),
                (Integer) arguments.get("expiryMonth"),
                (Integer) arguments.get("expiryYear"),
                (String) arguments.get("cvc")
        );
        Card card = builder.build();

        if (!card.validateCard()) {
          result.error("invalidCardData", "Invalid Card Data", null);
        } else {
          Stripe stripe = new Stripe(r == null ? binding.getApplicationContext() : r.activeContext(), (String) arguments.get("publishableKey"));
          stripe.createCardToken(card, new ApiResultCallback<Token>() {
            @Override
            public void onError(Exception error) {
              result.error("tokenizationException", "Exception creating token", error);
            }

            @Override
            public void onSuccess(Token token) {
              result.success(token.getId());
            }
          });

        }
        break;
      default:
        result.notImplemented();
        break;
    }
  }
}

File: ios/Classes/FlutterStripePlugin.m
#import "FlutterStripePlugin.h"
@import Stripe;

@implementation FlutterStripePlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"flutter_stripe"
            binaryMessenger:[registrar messenger]];
  FlutterStripePlugin* instance = [[FlutterStripePlugin alloc] init];
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  } else if ([@"getToken" isEqualToString:call.method]) {
      NSString* key = call.arguments[@"publishableKey"];
      STPAPIClient* client = [[STPAPIClient alloc] initWithPublishableKey:key];
      STPCardParams* cardParams = [STPCardParams alloc];
      cardParams.number = call.arguments[@"cardNumber"];
      NSNumber* year = call.arguments[@"expiryYear"];
      NSNumber* month = call.arguments[@"expiryMonth"];
      cardParams.expYear = [year intValue];
      cardParams.expMonth = [month intValue];
      cardParams.cvc = call.arguments[@"cvc"];
      STPCardValidationState state = [STPCardValidator validationStateForCard:cardParams];
      if (STPCardValidationStateInvalid == state) {
          result([FlutterError errorWithCode:@"invalidCardData" message:@"Invalid Card Data" details:nil]);
      } else {
          [client createTokenWithCard:cardParams completion:^(STPToken * _Nullable token, NSError * _Nullable error) {
              if (token == nil) {
                  result([FlutterError errorWithCode:@"tokenizationException" message:@"Exception creating token" details:nil]);
              } else {
                  result(token.tokenId);
              }
          }];
      }
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end
